### ISSUE TITLE
Changing a value of an item inside an ArraySchema is not sent to the client

### ISSUE THREAD

**Issue by DimosthenisK** [colyseus/colyseus#368]
Trying to change the value of an item inside an ArraySchema changes the state correctly, but the client doesn't get the change in the onStateChange listener.

```Typescript
execute({ sessionId, cardIndex }: this['payload']) {
    const player = this.state.players[sessionId] as Models.Player;
    player.hand.cards[cardIndex].isSelected = true;
    console.log(player.hand.cards[cardIndex].isSelected); //true as expected, still false in client
  }
```
I tried both with and without filters on version 0.14.

**Comment by endel** [comment]
Hi @DimosthenisK, thanks for trying out the version 0.14 alpha.

I've tried and failed to reproduce this issue you described though. Can you fork this and try to break it as it does in your project? https://github.com/endel/colyseus-issue-368

Thanks!

**Comment by DimosthenisK** [comment]
Hi @endel ! Thanks for taking the time to check out my issue.

I am very very sorry, I meant 0.13. I tried migrating to 0.14 but something wasn't working correctly and I had to drop it. 

Is it possible that 0.14 fixes this issue? Should I try migrating again?

**Comment by endel** [comment]
You're welcome, I see, that's very likely! Feel free to try it out and let me know if 0.14 works for you! It is going to be released this week anyway :)

**Comment by DimosthenisK** [comment]
That's awesome. Thank you!

I'll go ahead and try migrating again if it is going to be released so soon.

**Comment by DimosthenisK** [comment]
Problem was indeed solved by migrating to 0.14!
